### PR TITLE
Hetionet reconstruct

### PR DESCRIPTION
**Pull request by danich1**
This pull request adds the last two experiment results - Calibrating the discriminator model and recalling/adding edges to Hetionet.

### CODE REVIEW

File: compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py [hunk 1]
```diff
@@ -0,0 +1,330 @@
+
+# coding: utf-8
+
+# # Compound Treats Disease Edge Prediction
+
+# This notebook is designed to take the next step moving from predicted sentences to edge predictions
```
Comment (ajlee21) [inline compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py]: I would maybe be a bit more specific about what you mean by moving from predicting sentences to predicting edges

File: compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+import pandas as pd
+from sklearn.metrics import precision_recall_curve, roc_curve, auc
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+
+# In[2]:
+
+
+total_candidates_df = (
```
Comment (ajlee21) [inline compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py:24]: Maybe add a short comment to describe what each df is.  I couldn't really tell from the name, I could only tell after looking at the output

File: compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+# In[9]:
+
+
+performance_map = {}
+
+
+# In[10]:
+
+
+precision, recall, pr_threshold = precision_recall_curve(
```
Comment (ajlee21) [inline compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py:116]: So this curve is showing the performance of the prediction using the max, mean, median of what score exactly?

File: compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+)
+
+fpr, tpr, roc_threshold = roc_curve(
+    grouped_candidates_pred_df.hetionet, 
+    grouped_candidates_pred_df.model_prediction_median,
+)
+
+performance_map['median'] = {
+    "precision":precision, "recall":recall, 
+    "pr_threshold":pr_threshold, "false_pos":fpr,
```
Comment (ajlee21) [inline compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py:168]: What is the definition of this pr_threshold?  How is it determined?

File: compound_disease/compound_treats_disease/model_calibration_experiment/model_calibration_experiment.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
+
+# coding: utf-8
+
+# In[1]:
+
+
```
Comment (ajlee21) [inline compound_disease/compound_treats_disease/model_calibration_experiment/model_calibration_experiment.py:6]: I'm not sure what this nb is doing.  What are you calibrating?

**Review by ajlee21 — APPROVED** [review]
Looks like you've accomplished a lot of work, just think it might be helpful to add a few more comments places to clarify what things are.

File: compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py (hunk 3 above)
Comment (danich1) [reply compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py:116]: So each mention pair has a change of sharing a particular relationship. After my classification pipeline, each sentence that contains the given pair has a confidence score for establishing existence. A pair can have multiple sentences, so combining each score is unintuitive (i.e. is taking the max score appropriate here? how about median or mean?). The plot here shows the performance for doing each operation (max, mean, median) separately. Based on the results Max score seems the most appropriate.

File: compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py (hunk 4 above)
Comment (danich1) [reply compound_disease/compound_treats_disease/edge_prediction_experiment/edge_prediction_experiment.py:168]: pr_threshold -> precision recall threshold. This is determined through sklearns `precision_recall_curve` function.

File: compound_disease/compound_treats_disease/model_calibration_experiment/model_calibration_experiment.py (hunk 5 above)
Comment (danich1) [reply compound_disease/compound_treats_disease/model_calibration_experiment/model_calibration_experiment.py:6]: This part comes in after training the discriminative model. I'm using a deep learning model and deep learning models aren't calibrated correctly ([paper here](https://arxiv.org/pdf/1706.04599.pdf)). Based on this fact, I'm using a calibrator to make my predictions more trustworthy.

**Pull request merged**